CONTRIBUTING.md

```diff
@@ -0,0 +1,393 @@
+# Contributing to PyNumDiff
+
+Thank you for your interest in contributing to PyNumDiff! This document provides guidelines and instructions for contributing to the project. Following these guidelines helps communicate that you respect the time of the developers managing and developing this open-source project.
```

Comment:
lol, I wouldn't emphasize our need for respect. Anyone taking an interest is enough for us. Makes us sound a bit jilted and stiff, although maybe there's some deep truth to that for academics.